frontend/src/pages/signup/components/SignupForm/SignupForm.tsx

```diff
@@ -52,28 +65,107 @@ function SignupForm() {
 
   const phoneNumberErrorMessage = getPhoneNumberErrorMessage(phoneNumber);
 
+  const {
+    shouldBlockSubmitByPhoneNumberCheck,
+    handleAuthCodeClick,
+    getFinalPhoneNumberErrorMessage,
+    isPhoneNumberCheck,
+    isRequestCompleted,
+  } = useVerificationCodeRequest({ phoneNumber, phoneNumberErrorMessage });
+
   const {
     verificationCode,
     handleVerificationCodeChange,
     errorMessage: verificationCodeErrorMessage,
     isValid: isVerificationCodeValid,
   } = useVerificationCodeInput();
 
+  const {
+    verificationCodeError,
+    handleAuthCodeVerifyClick,
+    getFinalVerificationCodeErrorMessage,
+    shouldBlockSubmitByVerificationCode,
+  } = useVerificationCodeConfirm({
+    verificationCode,
+    verificationCodeErrorMessage,
+  });
+
   const validateForm = () => {
     const validations = [
       isNameValid,
-      isUserIdValid,
+      isUserIdValid && !duplicateError,
       isPasswordValid,
       isPasswordConfrimValid,
       phoneNumber !== '' && phoneNumberErrorMessage === '',
-      isVerificationCodeValid,
+      isVerificationCodeValid && !verificationCodeError,
     ];
 
     return validations.every(Boolean);
   };
 
+  const getVerificationButtonEnabled = () => {
+    return (
+      isPhoneNumberCheck &&
+      phoneNumberErrorMessage === '' &&
+      isVerificationCodeValid &&
+      isRequestCompleted
+    );
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (shouldBlockSubmitByUserId()) {
+      return;
+    }
+
+    if (shouldBlockSubmitByVerificationCode()) {
+      return;
+    }
+
+    if (shouldBlockSubmitByPhoneNumberCheck()) {
+      return;
+    }
+
+    const form = e.target as HTMLFormElement;
+    const data = new FormData(form);
+
+    const name = data.get('name');
+    const gender = data.get('gender');
+    const password = data.get('password');
+    const loginId = data.get('id');
+    const phone = data.get('phone');
+
+    if (
+      typeof name !== 'string' ||
+      (gender !== '남' && gender !== '여') ||
+      typeof password !== 'string' ||
+      typeof loginId !== 'string' ||
+      typeof phone !== 'string'
+    ) {
```

Comment:
p3: 이미지 파일을 다루거나 formData를 가공할 일이 없으니까
 formData에서 값을 꺼내지 않고 이미 각 훅에서 꺼낸 데이터를 바로 쓰면 좀 더 간단하게 바뀔 거 같아! 👍

```suggestion
  const invalidSignupInfo =
      !name ||
      (gender !== '남' && gender !== '여') ||
      !password ||
      !userId ||
      !phoneNumber;

    if (invalidSignupInfo) {
      return;
    }
```

Reply:
처음에는 상태를 바깥에 두지 않을 생각을 하고 submit함수를 구성하다 보니 저렇게 됐네.....😇 수정할게!!

Reply:
[refactor: 불필요한 formData 가공없이 state사용으로 변경](https://github.com/woowacourse-teams/2025-Fit-toring/pull/274/commits/1f7274bd29510ca7ef4e153b494c52b1bdb9bbbe)